gratipay/security/csrf.py

```diff
@@ -99,32 +87,6 @@ def get_csrf_token_from_request(request):
     # Assume that anything not defined as 'safe' by RC2616 needs protection
     if request.line.method not in ('GET', 'HEAD', 'OPTIONS', 'TRACE'):
 
-        if _is_secure(request):
-            # Suppose user visits http://example.com/
-            # An active network attacker (man-in-the-middle, MITM) sends a
-            # POST form that targets https://example.com/detonate-bomb/ and
-            # submits it via JavaScript.
-            #
-            # The attacker will need to provide a CSRF cookie and token, but
-            # that's no problem for a MITM and the session-independent
-            # nonce we're using. So the MITM can circumvent the CSRF
-            # protection. This is true for any HTTP connection, but anyone
-            # using HTTPS expects better! For this reason, for
-            # https://example.com/ we need additional protection that treats
-            # http://example.com/ as completely untrusted. Under HTTPS,
-            # Barth et al. found that the Referer header is missing for
-            # same-domain requests in only about 0.2% of cases or less, so
-            # we can use strict Referer checking.
-            referer = request.headers.get('Referer')
-            if referer is None:
-                raise Response(403, REASON_NO_REFERER)
-
-            good_referer = 'https://%s/' % _get_host(request)
-            if not same_origin(referer, good_referer):
```

Comment:
Where is `same_origin` coming from? I'm not seeing that this PR removes a definition or import for it. Are we using it elsewhere?